feat(2645): Add AWS Secrets Manager wrapper class

jimwashbrook-ons opened the pull request:
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

### What

Adds a class that encapsulates the logic of fetching secrets from AWS Secrets Manager.

### Testing

- [x] Yes
- [ ] No
Please write a brief description of why test coverage is not necessary here.
- [ ] Not as part of this ticket. (Could be done at a later point)

100% coverage.

### Documentation

- [x] Yes
- [ ] No
Please write a brief description of why documentation is not necessary here.
- [ ] Not as part of this ticket. (Could be done at a later point)

Added brief overview in a README.

### How to review

#### Single secret

1. [Create a secret in AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/create_secret.html). This can also be done using the AWS CLI, _and_ can be done completely locally using Localstack. See the [/tests/secrets/README.md](https://github.com/ONSdigital/dp-python-tools/blob/feat/2645/aws-secrets-manager-fetcher/tests/secrets/README.md) for a very basic overview.
2. Instantiate an instance of the [SecretsClient](/dpytools/secrets/secrets_client.py) class
3. Retrieve the secret from step 1 by calling instantiated_class_object.get_secret(SECRET_ID)

E.g. (where the secret name you created in step 1 is `example_secret` with value `secret value`)
```python
from dpytools.secrets.secrets_client import SecretsClient

client = SecretsClient()
response = client.get_secret("example_secret")

assert response.error is None
assert response.success is True
assert response.value == "secret value"
```

#### Multiple secrets (batch)

1. Follow step 1 in the last section (Single Secret), but create _multiple_ secrets
2. Instantiate an instance of the [SecretsClient](/dpytools/secrets/secrets_client.py) class
3. Retrieve the secrets from step 1 by calling instantiated_class_object.batch_get_secrets(SECRET_IDS)

E.g. (where the secret ids you created in step 1 are `["example_one", "id_two"]`

```python
from dpytools.secrets.secrets_client import SecretsClient

client = SecretsClient()
response = client.batch_get_secrets(["example_one", "id_two"])

assert response.error is None
assert response.success is True
assert len(response.secrets) == 2
# Other assertions here
```

jimwashbrook-ons committed:
feat(secrets): AWS Secrets Manager wrapper to get a single secret

feat(secrets): Add wrapper for AWS Secrets Manager

tests(secrets): Remaining tests for exceptions

docs(secrets): Add README

chore: Update version

chore: Formatting + linting

chore: Use typings module instead of later python functionality

chore: iosort

chore: iosort on tests

chore: correct version

feat: Add logic for handling JSON + binary secrets better

refactor: Try set inner value from JSON, tweak methods slightly

SarahJohnsonONS reviewed (commented):
All good, except the `value` is a string of the key-value pair instead of just the value (i.e. `'{"sj-test-secret":"this_is_sarahs_secret"}'`)

jimwashbrook-ons committed:
feat: batch get secrets

chore: docs

chore: functional testing

chore: docstrings + add a couple fields

tests: Update tests + fix errors

chore: rename to match method

tests: add helpers

tests: batch tests WIP

tests: batch tests WIP

tests: 100% coverage

chore: const string for key

chore: linting + formatting

docs: add examples to readme

SarahJohnsonONS commented on dpytools/secrets/secret.py:
```diff
@@ -0,0 +1,64 @@
+    ):
+        self._aws_value = value
+        self.error = error
+        self.success = self.value is not None and error is None
```
Getting an `AttributeError` here when instantiating the `Secret`: `'Secret' object has no attribute 'value'`

jimwashbrook-ons replied:
Should be fixed now :)

jimwashbrook-ons commented:
> All good, except the `value` is a string of the key-value pair instead of just the value (i.e. `'{"sj-test-secret":"this_is_sarahs_secret"}'`)

I've added some handling around JSON now. Probably not ideal, but it should allow easier handling of JSON responses.

SarahJohnsonONS reviewed (approved):
LGTM
